src/ResourceManager/RecoveryServices.SiteRecovery/Commands.RecoveryServices.SiteRecovery/Vault/SetAzureRmRecoveryServicesAsrVaultContext.cs

```diff
@@ -0,0 +1,133 @@
+// ----------------------------------------------------------------------------------
+// 
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Net;
+using System.Management.Automation;
+using Microsoft.Rest.Azure;
+using Microsoft.Azure.Portal.RecoveryServices.Models.Common;
+using Microsoft.Azure.Commands.RecoveryServices.SiteRecovery.Properties;
+using Microsoft.Azure.Management.RecoveryServices.Models;
+
+namespace Microsoft.Azure.Commands.RecoveryServices.SiteRecovery
+{
+    /// <summary>
+    ///     Sets the Recovery Services vault context to be used for subsequent Azure Site Recovery operations in the current PowerShell session.
+    /// </summary>
+    [Cmdlet(
+        VerbsCommon.Set,
+        "AzureRmRecoveryServicesAsrVaultContext",
+        DefaultParameterSetName = ASRParameterSets.ARSVault,
+        SupportsShouldProcess = true)]
+    [Alias(
+        "Set-ASRVaultContext",
+        "Set-ASRVaultSettings",
+        "Set-AzureRmRecoveryServicesAsrVaultSettings")]
+    [OutputType(typeof(ASRVaultSettings))]
+    public class SetAzureRmRecoveryServicesAsrVaultSettings : SiteRecoveryCmdletBase
+    {
+        /// <summary>
+        ///     Gets or sets the Recovery Services vault object corresponding to the Recovery Services vault.
+        /// </summary>
+        [Parameter(
+            ParameterSetName = ASRParameterSets.ARSVault,
+            Mandatory = true,
+            ValueFromPipeline = true)]
+        [ValidateNotNullOrEmpty]
+        public ARSVault Vault { get; set; }
+
+        /// <summary>
+        ///     ProcessRecord of the command.
+        /// </summary>
+        public override void ExecuteSiteRecoveryCmdlet()
+        {
+            base.ExecuteSiteRecoveryCmdlet();
+
+            if (this.ShouldProcess(
+                this.Vault.Name,
+                VerbsCommon.Set))
+            {
+                switch (this.ParameterSetName)
+                {
+                    case ASRParameterSets.ARSVault:
+                        this.SetARSVaultContext(this.Vault);
+                        break;
+                    default:
```

Comment:
You don't need this, powershell will never allow the user to enter an invalid parameter set.

Reply:
sure.